app.rb

```diff
@@ -0,0 +1,98 @@
+
+module Enumerable
+  def my_each(&prc)
+    length.times do |i|
+      prc.call(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index(&prc)
+    length.times do |i|
+      prc.call(self[i], i)
+    end
+    self
+  end
```

Comment:
- [ ] `my_each_with_index` is not working as the original `each_with_index` method, also is not passing all test cases.

### `each_with_index` is an Enumerable that has these characteristics: 
- Returns an Enumerator if no block is given.
```
[1, 2, 3].each_with_index
# #<Enumerator: [1, 2, 3]:each_with_index>
````
- it returns the array/range/hash itself if a block is given.
```
[1, 2, 3].each_with_index { |num, idx| num }
# [1, 2, 3]
(1..3).each { |num, idx| num * idx }
# 1..3
````
- it applies the block for each element inside the array/range/hash. It's failing with ranges and hashes
```
[1, 2, 3].each_with_index { |num, idx| puts num, idx }
# 1
# 0
# 2
# 1
# 3
# 2
# => [1, 2, 3]
```
[Enumerator#each](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-each_with_index)